swarm.go

```diff
@@ -115,6 +116,13 @@ func NewSwarm(ctx context.Context, local peer.ID, peers peerstore.Peerstore, bwc
 	s.proc = goprocessctx.WithContextAndTeardown(ctx, s.teardown)
 	s.ctx = goprocessctx.OnClosingContext(s.proc)
 
+	// TODO register provider for introspection
+	if introspector != nil {
```

Comment:
Would love to see the actual provider functions, after my comments on https://github.com/libp2p/go-libp2p-core/pull/107 have been integrated.